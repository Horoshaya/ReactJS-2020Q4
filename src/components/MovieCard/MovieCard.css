.card {
  width: 300px;
  color: #ffffff;
}

.img {
  width: 100%;
}

.mainInfo {
  display: flex;
  justify-content: space-between;
}

.year {
  padding: 2px 10px;
  border: 1px solid #ffffff;
  border-radius: 3px;
}
