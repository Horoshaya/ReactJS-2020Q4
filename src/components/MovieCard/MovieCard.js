import React from "react";
import PropTypes from "prop-types";

const MovieCard = ({ title, description, year }) => {
  return (
    <div>
      <h3>{title}</h3>
      <p>{description}</p>
      <p>{year}</p>
    </div>
  );
};

MovieCard.PropTypes = {
  title: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
};

export default MovieCard;
