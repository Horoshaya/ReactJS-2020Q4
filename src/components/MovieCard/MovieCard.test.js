import React from 'react';
import rerender from 'react-test-renderer';
import MovieCard from './MovieCard';
import { BrowserRouter as Router } from 'react-router-dom';

describe('when NotFound', () => {
  test('then snapshot created', () => {
    const component = rerender.create(
      <Router>
        <MovieCard />
      </Router>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
