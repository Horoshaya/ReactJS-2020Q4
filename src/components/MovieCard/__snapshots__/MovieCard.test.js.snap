// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when NotFound then snapshot created 1`] = `
<div>
  <div
    onClick={[Function]}
  />
  <img
    onError={[Function]}
  />
  <div>
    <h3 />
    <p>
      NaN
    </p>
  </div>
  <p />
  <p />
</div>
`;
